### DISCUSSION TITLE
[Paperless-NGX] OCR not working with additional languages

### DISCUSSION THREAD

**Discussion by Stooovie** [tteck/Proxmox#3012] [Help Needed]
I need to add Czech (CES) to OCR because I'm getting `MissingDependencyError: OCR engine does not have language data for the following requested languages: CES`

PNGX documentation says:
`If you run paperless on docker, paperless.conf is not used. Rather, configure paperless by copying necessary options to docker-compose.env`

LXC is not Docker, so I'm editing /opt/paperless/paperless.conf 

Adding `PAPERLESS_OCR_LANGUAGE=eng ces` makes zero difference.

anything else? thanks!

**Comment by sbstn-0x2a** [comment]
in lxc
`apt install tesseract-ocr-ces`

restart paperless service

**Reply by Stooovie** [reply]
Sorry but didn't help (package did install correctly). Even after reboot, same error message. `ces` is the correct code for Czech.

**Reply by sbstn-0x2a** [reply]
You can look at 
`tail -n50 /opt/paperless/data/log/paperless.log`

or on the paperless website under protocol point.

**Reply by sbstn-0x2a** [reply]
look at the paperless.conf under PAPERLESS_OCR_LANGUAGE.
Look for `:` or `=`
In Paperless LXC you find config under `/opt/paperless/paperless.conf`
It must be an `=` 
Test it with a single language first:
`PAPERLESS_OCR_LANGUAGE=ces`
After it works, test with two languages.
`PAPERLESS_OCR_LANGUAGE=eng+ces`

Link to Paperless documentation:
[https://docs.paperless-ngx.com/configuration/#PAPERLESS_OCR_LANGUAGE](https://docs.paperless-ngx.com/configuration/#PAPERLESS_OCR_LANGUAGE)

**Reply by Stooovie** [reply]
Sadly, not working. Still queued indefinitely (well, 1-page PDF for 20 minutes). I am not getting the error message anymore (MissingDependencyError: OCR engine does not have language data for the following requested languages: CES), but nothing's happening. I have restarted the LXC after each adjustment.

Thanks for your help but this is where I give up, I have already too many systems to babysit 🥲

**Reply by sbstn-0x2a** [reply]
its ok.
if you want to try it again, give the lxc the power to ocr.
ocr is cpu intense, if you have 2 languages, you need 2 times longer.
personally i have set 16 cores to paperless and 8gb ram ocr my documents.
the default setting from ttecks script is to low for my opinion. only for running the lxc without doing anything ;)

```
PAPERLESS_TASK_WORKERS=4
PAPERLESS_THREADS_PER_WORKER=4
```

[https://docs.paperless-ngx.com/configuration/#PAPERLESS_TASK_WORKERS](https://docs.paperless-ngx.com/configuration/#PAPERLESS_TASK_WORKERS)

greets toxo

**Comment by Stooovie** [comment]
I started from scratch, no dice.

With default settings, PNGX complains (even with notification in GUI) about `MissingDependencyError: OCR engine does not have language data for the following requested languages: ces`

Uncommenting and adding
`PAPERLESS_OCR_LANGUAGE=ces`

results in the file stuck in Queued (NO GUI warning)
with 

```
Error occurred while consuming document mrdat.pdf: MissingDependencyError: OCR engine does not have language data for the following requested languages:
ces
Please install the appropriate language data for your OCR engine.
```


in log.

installing `apt install tesseract-ocr-ces` inside the LXC finishes fine, leads to the exact same behaviour.

I don't think this is me. CPU should not be an issue, the OCR process is not starting with Czech language at all.

**Comment by Stooovie** [comment]
It seems just installing the tesseract-ocr-ces package and not touching the conf file at all at the very least allows for the scanning to start and finish. The scanned document looks fine. Let's see what happens when I add english PDF.

**Reply by Snowii123** [reply]
hi there, how did this work out at the end please? thinkiong of installing paperless-nxg too but it working with Czech is essential for me... thank you!
